### PR TITLE
docs(interfaces): document important interfaces and classes

### PR DESCRIPTION
**Pull request by staltz**
Write JSDoc documentation for Observer, Subscriber, Subscription, Notification, Scheduler, Action,
etc.

### CODE REVIEW

File: src/Observer.ts [hunk 1]
```diff
@@ -21,6 +21,68 @@ export interface CompletionObserver<T> {
...
+ * call the Observer's `next(value)` method to provide notifications. A
+ * well-behaved Observable will call an Observer's `complete()` method exactly
+ * once or the Observer's `error(err)` method exactly once, as the last
+ * notification delivered.
+ *
+ * @interface
+ * @name Observer
+ * @noimport true
+ */
+export class ObserverDoc<T> {
```
Comment (blesh) [inline src/Observer.ts]: interesting, does this build as part of the output? That wouldn't be ideal.

Too bad there isn't a way to document interfaces. Perhaps we add some files to the build that are purely for documenting interfaces and build them separately?

Comment (staltz) [inline src/Observer.ts]: Yeah it's ugly, I know.

> Perhaps we add some files to the build that are purely for documenting interfaces and build them separately?

We can do that. The real reason is that TypeScript strips away JSDoc comments for interfaces. And I won't try another PR to TS in vain if they are so... 'courteous'.

Comment (kwonoj) [inline src/Observer.ts]: Agree it's pain to deal with comment strip in typescript with docs. :/

**Comment by staltz** [comment]
I rebased and also moved interface documentations to files `MiscJSDoc.ts` which are not compiled to any release bundle, but appear in the ESDoc site.

Can haz merge now?

cuteness

**Comment by blesh** [comment]
LGTM.

**Comment by staltz** [comment]
Rebased. :white_check_mark:

**Comment by kwonoj** [comment]
Merged with 8da7a8f, thanks @staltz :)

**Pull request merged**